receiver/statsdreceiver/internal/protocol/statsd_parser.go

```diff
@@ -410,11 +410,18 @@ func parseMessageToMetric(line string, enableMetricType bool) (statsDMetric, err
 
 			for _, tagSet := range tagSets {
 				tagParts := strings.SplitN(tagSet, ":", 2)
-				if len(tagParts) != 2 {
-					return result, fmt.Errorf("invalid tag format: %s", tagParts)
-				}
 				k := tagParts[0]
-				v := tagParts[1]
+				if k == "" {
+					return result, fmt.Errorf("invalid tag format: \"%s\"", tagSet)
```

Comment:
i updated the error format b/c the built-in list serializer is a bit hard to identify errors in, e.g. `[ val1]` is what will happen when you have an empty key. `":val1"` seems easier to parse/understand.

Reply:
```suggestion
					return result, fmt.Errorf("invalid tag format: %q", tagSet)
```
This will quote the value.

Reply:
ah thank you! i need to learn more about the format tokens.